zpay32/invoice.go

```diff
@@ -118,6 +123,10 @@ type Invoice struct {
 	// invoice.
 	PaymentHash *[32]byte
 
+	// PaymentAddr is the payment address to be used by payments to prevent
+	// probing of the destination.
+	PaymentAddr *[32]byte
```

Comment:
In another PR, I experimented with a strong type for this. What do you think of that?

Reply:
hmm if we go with a strong type it would hinder the consolidation of logic in this pr. my thought was that a type here doesn't offer that much as it's just a static byte array with no hashes/preimages, but it could be cast into a proper type elsewhere in the codebase if necessary. thoughts?

Reply:
Maybe you could do `([32]byte, error)` as the return type of `parse32Bytes` and directly cast it into the `Invoice` field? I am fine either way, but noticed that I do appreciate the stricter types.